vpr/src/device/rr_graph_builder.h

```diff
@@ -235,6 +280,21 @@ class RRGraphBuilder {
     /* Fast look-up for rr nodes */
     RRSpatialLookup node_lookup_;
 
+    /* Segment relatex data
+     * Segment info should be corrected annotated for each rr_node
+     * whose type is CHANX and CHANY
+     */
+    vtr::vector<RRSegmentId, t_segment_inf> rr_segments_; /* detailed information about the segments, which are used in the RRGraph */
+    vtr::vector<RRSegmentId, RRSegmentId> segment_ids_;   /* unique identifiers for routing segments which are used in the RRGraph */
+    /* Switch related data
```

Comment:
As requested by @vaughnbetz , we should enrich the comments here to guide developers:

```
 /* Autogenerated in build_rr_graph based on switch fan-in.
  *  - Each rr_switch contains the detailed information of a routing switch interconnecting two routing resource nodes. 
  *  - We use a fly-weight data structure here, in the same philosophy as the rr_indexed_data. See detailed explanation in the t_rr_switch_inf data structure
  */
```